packages/story-editor/src/components/floatingMenu/menus/product.js

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { __ } from '@googleforcreators/i18n';
+import { memo, useCallback } from '@googleforcreators/react';
+
+/**
+ * Internal dependencies
+ */
+import { Separator, Dismiss, Trash } from '../elements';
+import { useStory } from '../../../app/story';
+import ProductDropdown from '../../library/panes/shopping/productDropdown';
+import { useFloatingMenu } from '../context';
+
+const FloatingProductMenu = memo(function FloatingProductMenu() {
+  const { handleDismiss } = useFloatingMenu(
+    ({ actions: { handleDismiss } }) => ({
+      handleDismiss,
+    })
+  );
+
+  const { selectedElement, updateElementById, deleteElementById } = useStory(
+    ({ actions, state }) => ({
+      updateElementById: actions.updateElementById,
+      selectedElement: state.selectedElements[0],
+      deleteElementById: actions.deleteElementById,
+    })
+  );
+
+  const setProduct = useCallback(
+    (product) => {
+      updateElementById({
+        elementId: selectedElement.id,
+        properties: { product },
+      });
+    },
+    [selectedElement, updateElementById]
+  );
+
+  return (
+    <>
+      <ProductDropdown
+        product={selectedElement?.product}
+        setProduct={setProduct}
+      />
+      <Separator />
+      <Trash
+        title={__('Remove product', 'web-stories')}
+        handleRemove={() => {
+          deleteElementById({ elementId: selectedElement.id });
```

Comment:
There's a dedicated `deleteSelectedElements` reducer we can use here.

It does not make sense to render this dropdown for multi-selection, so we should avoid that. Then this reducer is safe to use.